packages/services/f-braze-adapter/src/services/BrazeConsumerRegistry.js

```diff
@@ -0,0 +1,132 @@
+            userId,
+            enableLogging
+        });
+
+        return registryInstance;
+    }
+
+    get dispatcher () {
+        return this._dispatcher;
+    }
+
+    constructor (dispatcher) {
+        this.consumers = [];
+
+        // subscribe to event stream for braze cards and messages
+        dispatcherEventStream.subscribe(CONTENT_CARDS_EVENT_NAME, this.applyContentCardCallbacks);
+        dispatcherEventStream.subscribe(IN_APP_MESSAGE_EVENT_NAME, this.applyInAppMessageCallbacks);
+        dispatcherEventStream.subscribe(IN_APP_MESSAGE_EVENT_CLICKS_NAME, this.applyInAppMessageClickEventsCallbacks);
+        dispatcherEventStream.subscribe(LOGGER, log => this.applyLoggerCallbacks(...log));
+
+        this._dispatcher = dispatcher;
+    }
+
+    /**
+     * For each consumers logger callbacks fire the callback with log data
+     * @param type
+     * @param message
+     * @param data
+     */
+    applyLoggerCallbacks (type, message, data) {
+        this.consumers.reduce((acc, consumer) => [...acc, ...consumer.getLoggerCallbacks()], [])
+            .forEach(callback => {
+                callback[type](message, data);
+            });
+    }
+
+    /**
+     * For each consumers content card callbacks fire the callback with supplied cards
+     * @param cards
+     */
+    applyContentCardCallbacks (cards) {
+        this.consumers.reduce((acc, consumer) => [...acc, ...consumer.getContentCardCallbacks()], [])
+            .forEach(callback => {
+                callback(cards);
+            });
+    }
+
+    /**
+     * For each consumers in app message callbacks fire the callback with the supplied message
+     * @param message
+     */
+    applyInAppMessageCallbacks (message) {
+        this.consumers.reduce((acc, consumer) => [...acc, ...consumer.getInAppMessagesCallbacks()], [])
+            .forEach(callback => {
+                callback(message);
+            });
+    }
+
+    /**
+     * For each consumers in app message click callbacks fire the callback with the supplied message
+     * @param message
+     */
+    applyInAppMessageClickEventsCallbacks (message) {
+        this.consumers.reduce((acc, consumer) => [...acc, ...consumer.getInAppMessageClickEventCallbacks()], [])
+            .forEach(callback => {
+                callback(message);
+            });
+    }
+
+    /**
+     * Register a consumer instance in the registry
+     * @returns {BrazeConsumer}
+     */
+    register (options) {
+        const consumer = new BrazeConsumer(options);
+        this.consumers.push(consumer);
+        return consumer;
+    }
+
+    /**
+     * Remove the passed consumer from the registry
+     * @param consumer {BrazeConsumer}
+     */
+    unregister (consumer) {
+        if (this.consumers.findIndex(c => c === consumer) !== -1) {
+            this.consumers.splice(this.consumers.findIndex(c => c === consumer), 1);
+        } else {
+            throw new Error("Can't find consumer in registry");
```

Comment:
Could the message be a bit more descriptive?  e.g. _Failed to unregister consumer, consumer not in registry_.

Reply:
replaced with your good suggestion